### ISSUE TITLE
Installing Fairseq2 fails on M1 Pro

### ISSUE THREAD

**Issue by M2LabOrg** [facebookresearch/fairseq2#211]
@cbalioglu created v0.1.1, as described in issue #1 . I have been trying to install it on a Macbook (M1 Pro) in a fresh environment, but I still get an error. Since that issue is closed, I opened a new one. I have been trying to install it for days without any success. I need to use Seamless models from Meta, which require Fairseq2. 

Any suggestions?

```
$ pip install fairseq2==0.1.1
Collecting fairseq2==0.1.1
  Using cached fairseq2-0.1.1-py3-none-any.whl.metadata (1.2 kB)
INFO: pip is looking at multiple versions of fairseq2 to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement fairseq2n==0.1.1 (from fairseq2) (from versions: none)
ERROR: No matching distribution found for fairseq2n==0.1.1
```

And:

```
$ pip install fairseq2
Collecting fairseq2
  Using cached fairseq2-0.2.0-py3-none-any.whl.metadata (1.2 kB)
INFO: pip is looking at multiple versions of fairseq2 to determine which version is compatible with other requirements. This could take a while.
  Using cached fairseq2-0.1.1-py3-none-any.whl.metadata (1.2 kB)
  Using cached fairseq2-0.1.0-py3-none-any.whl.metadata (1.2 kB)
ERROR: Cannot install fairseq2==0.1.0, fairseq2==0.1.1 and fairseq2==0.2.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    fairseq2 0.2.0 depends on fairseq2n==0.2.0
    fairseq2 0.1.1 depends on fairseq2n==0.1.1
    fairseq2 0.1.0 depends on fairseq2n==0.1.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

**Comment by cbalioglu** [comment]
Hey @M2LabOrg, do you mind describing all the steps you perform to reproduce this issue? Including how you create the env and in which order you call the "pip install" commands? Also it might help to clear your pip cache with `pip cache purge`.

**Comment by M2LabOrg** [comment]
Hey @cbalioglu,
Thank you for your message. I have tried to install it using a new environment and with `pip cache purge`, as you suggested. I still get the same error. Here's what I did:

```
% bash
$ mkvirtualenv fairseq2
$ pip cache purge
$ pip install fairseq2==0.1.1
$ pip install --upgrade pip
$ pip install fairseq2==0.1.1
```


I have attached the output here:
<img width="1026" alt="Screenshot 2024-01-01 at 10 33 50" src="https://github.com/facebookresearch/fairseq2/assets/68891146/bf07c531-e0f6-4a49-bdff-8f1a09d03f71">

**Comment by M2LabOrg** [comment]
Hey @cbalioglu, 

I wanted to share with you that I successfully resolved my installation issue by following the suggestion of a friend who recommended using Miniforge via Homebrew. Here are the steps I took:

```bash
% brew uninstall conda
% brew install miniforge
% mamba create fseq2
% mamba init
% mamba activate fseq2
% pip install fairseq2
```

It appears that the problem may have been related to conflicts with the environment created by virtualenvwrapper or conflicts with conda. However, using Miniforge via Homebrew proved to be an effective solution.

Thanks for your assistance!
